12.0 add commown self troubleshooting theft and loss

loganSIO opened the pull request.

loganSIO committed:
Factor out run_tour method in tests

loganSIO committed:
Add product informations on contract selector of all ST pages

This will be useful for coming theft and loss page for instance.

loganSIO committed:
Explicit stage_ref in ST forms as hidden inputs
The module commown_self_troubleshooting must now be explicit.

loganSIO committed:
Add theft and loss dedicated ST page

We edit the structure of main controller by adding contractual_issue type and date.
We also implemented unit test for JS and Python.
Finally, we add Declared theft and loss columns in project.xml

loganSIO committed:
Explicit project_ref in ST forms as hidden inputs

The default project_ref was support_project. Now there is no more default value.
Also, we put factor out project_ref inputs in a single one on top of the form.

loganSIO commented:
Hello @fcayre , il faudrait ici m'aider à interpréter le codecov, il y a un truc que je n'arrive pas à saisir ici.

fcayre commented on commown_self_troubleshooting/data/common_steps.xml:
```diff
@@ -37,8 +37,13 @@
           sélectionnez ici le 1er contrat de la liste et précisez-nous
           en introduction de votre message que vous ne le trouvez pas,
-          nous nous en occuperons !
+          nous nous en occuperons !<br/><br/>
```
`<br>` c'est vraiment un markup html pourri (non porteur de sens, juste de la mise en forme donc rien à faire dans le document), à éviter. Il suffit de fermer le `<p>` et en rouvrir un autre, sémantiquement c'est pas idiot ici.

loganSIO replied:
Fait !

fcayre commented on commown_self_troubleshooting/data/common_steps.xml:
```diff
@@ -37,8 +37,13 @@
+
+          <b>Vigilance :</b> Pour un vol ou une perte inscrivez le numéro de série / IMEI sur une feuille,
+          car vous devrez l'indiquer à la gendarmerie lors de l'établissement de plainte pour le vol
+          et/ou à votre opérateur pour le blocage du téléphone.
```
C'est très moche ça, de mettre un message utile quand dans un cas dans tous les AD... On peut pas intégrer dans le parcours / dans le texte du ticket / dans le mail automatique l'IMEI ou le N/S ? Je suis sûr qu'il y a une soltuion à la fois simple et élégante.

fcayre commented on commown_self_troubleshooting/data/ContractManagement/theftandloss/form.xml:
```diff
@@ -50,7 +50,7 @@
 
           <div class="bg-white border m-0 p-3 rounded">
             <h1>Vol ou Perte</h1>
-            <h2>On m'a volé ou j'ai perdu mon appareil</h2>
```
Ce genre de modif il faut utiliser git commit --amend ou git rebase -i et l'instruction fixup pour que le relecteur n'ait pas à relire toutes les versions du texte d'une part, et que l'historique git ne soit pas alourdi par des changements inutiles.
Si c'est pas clair pour toi Luc et moi sommes à ta dispo pour t'expliquer.

loganSIO replied:
Fait !

fcayre replied:
Ça n'a pas l'air fait non, sinon je ne le verrai plus dans ce commit :)

loganSIO replied:
Fait

fcayre commented on commown_self_troubleshooting/data/ContractManagement/theftandloss/form.xml:
```diff
@@ -122,9 +122,11 @@
+                      mettre en œuvre pour vous accompagner dans les démarches et ainsi pouvoir
+                      vous renvoyer un autre appareil au plus vite ! :-)<br /><br />
+                      Dans un premier temps, dans le champ de texte ci-dessous, pourriez-vous me
+                      relater ce qui s'est exactement passé (où-comment-quand) ?<br />
```
idem

loganSIO replied:
Fait !

fcayre commented on commown_self_troubleshooting/data/ContractManagement/theftandloss/form.xml:
```diff
@@ -151,7 +153,8 @@
                       Veuillez renseigner dans le champ de texte ci-dessous
-                      la date présumée de la perte ainsi que les circonstances de celle-ci.
+                      la date présumée de la perte ainsi que les circonstances
+                      de celle-ci.
```
idem

fcayre replied:
re-idem

loganSIO replied:
Fait

fcayre commented on commown_self_troubleshooting/data/ContractManagement/theftandloss/form.xml:
```diff
@@ -185,7 +188,7 @@
     <field name="view_id" ref="theft-and-loss" />
     <field name="website_meta_title">Vol ou Perte</field>
-    <field name="website_meta_description">On m'a volé ou j'ai perdu mon appareil</field>
+    <field name="website_meta_description">Je me suis fait voler ou j'ai perdu mon appareil</field>
```
idem

loganSIO replied:
Fait !

fcayre replied:
idem

loganSIO replied:
Fait

fcayre commented on commown_self_troubleshooting/data/troubleshooting.xml:
```diff
@@ -125,6 +125,13 @@
+      <field name="requires_contract" eval="True"/>
+      <field name="contract_domain"></field>
+    </record>
+
```
Il faut amender le commit qui ajoute la page plutôt que d'ajouter cette modif dans un commit séparé.

loganSIO replied:
Fait

fcayre commented on commown_devices/tests/common.py:
```diff
@@ -1,6 +1,11 @@
 import dateutil.parser
 
-from odoo.addons.product_rental.tests.common import RentalSaleOrderTC
```
gnnii ? il faudrait pas plutôt que tu rebases ta PR sur la 12.0 ?

loganSIO replied:
Fait

fcayre reviewed (changes requested):
Il faut que tu gagnes en aisance avec git si nécessaire mais :
1. il semblerait que tu n'aies pas rebasé ta PR sur la 12.0 car ta dépendance aux modifs d'une autre PR est anormale
2. L'historique git doit aider ses futurs lecteurs à comprendre les grandes évolutions du code sans trop d'effort. Les modifs de texte sur des modifs de texte sont du bruit qui va à l'encontre de cet objectif.
3. Il faut faciliter la tâche du relecteur, dont le rôle est précieux pour garder une qualité de code élevée. Ça a (entre autres) mes mêmes conséquences que le point précédent : amender les commits précédents au lieu d'ajouter des commits qui nuisent à la compréhension du relecteur et lui rendent la tâche plus fastidieuse n'est pas une bonne idée. Pour ça il faut une maîtrise TOTALE du gestionnaire de version utilisé (git ici).

loganSIO commented:
Toutes les modifications demandées ont été effectuées suite au dernier push.

En effet, c'était très crado, c'est vraiment le problème de laisser trainer son dev aussi longtemps.
Je serais beaucoup plus vigilent à ce sujet désormais, merci pour l'aide apportée.

fcayre commented on commown_self_troubleshooting/data/ContractManagement/theftandloss/form.xml:
```diff
@@ -0,0 +1,189 @@
+                let $details = $('#contract_details').show();
+                $details.find('#contract_name').text(contractData.contractName);
+                $details.find('#contract_commitment_end_date').text(contractData.commitmentEndDate);
+
```
Il serait très facile ici d'afficher le n° de série / IMEI à l'utilisateur dans le texte quelque part, pour sa déclaration à la police.

loganSIO replied:
Fait!

fcayre replied:
Attention contractData.serial peut être vide : il faut alors masquer le html correspondant.

fcayre replied:
Pour moi ça le fait toujours pas : si le n° de série est vide, le message présenté aux utilisateurs ne convient pas (il y juste rien à la place du n° de série, ça le fait pas...). Je pense qu'il faut plutôt prévoir un message explicite et bien écrit, dans le html, pour le cas où il n'y a pas de n° de série et qu'on cache quand il y en a un, à coup de js, ci-dessus (avec un if / else complet donc).

fcayre commented on commown_self_troubleshooting/data/common_steps.xml:
```diff
@@ -68,7 +68,7 @@
           </div>
         </div>
-      </div>
+    </div>
```
Une erreur (à corriger) ici sans doute, à vérifier.

loganSIO replied:
Fait!

fcayre commented on commown_self_troubleshooting/data/ContractManagement/theftandloss/issue_template.xml:
```diff
@@ -9,9 +9,9 @@
+            <t t-if="theft_loss=='loss'">J'ai perdu mon appareil</t>
           </t>
         </p>
 
```
idem

loganSIO replied:
Fait

fcayre replied:
Fait ? On parlait bien d'afficher le n° de série ici non ? Je suis perdu, mais je ne le vois pas ici en tout cas !?

loganSIO replied:
Non, on l'affiche déjà dans le titre

fcayre reviewed (changes requested):
On y est presque, go go !

fcayre reviewed (changes requested):
Dernier message de commit à détailler.
Vérifier que les tests javascript sont exécutés.
Ajouter les mails templates et les project.task.type

loganSIO committed:
Add mail_templates.xml and update automatic mail in columns

Since we want to send a automatic e-mail to customer, we do need
to have a mail_templates.xml backup in the code whenever we migrate odoo version.

loganSIO commented:
Hello @fcayre les  points à faire ont été faits.

Review time !

fcayre commented on commown_self_troubleshooting/data/ContractManagement/theftandloss/form.xml:
```diff
@@ -0,0 +1,210 @@
+                let $details = $('.contract_details').show();
+                $details.find('#contract_name').text(contractData.contractName);
+                $details.find('#contract_commitment_end_date').text(contractData.commitmentEndDate);
+                $details.find('.contract_serial').text('(' + contractData.serial + ')' || '');
```
Tu traites mal le cas où contractData.serial est vide (ici tu te retrouverais avec des parenthèses vides), rajoute plutôt un if tout bête.

loganSIO replied:
Traité dans le dernier push

fcayre commented on commown_self_troubleshooting/data/ContractManagement/theftandloss/issue_template.xml:
```diff
@@ -0,0 +1,26 @@
+        <p>
+          <t t-if="theft_loss=='theft'">Mon appareil a été volé</t>
+          <t t-else="">
+            <t t-if="theft_loss=='loss'">J'ai perdu mon appareil</t>
```
utiliser plutôt t-elif, ce [la doc](https://www.odoo.com/documentation/12.0/developer/reference/qweb.html#conditionals)

loganSIO replied:
Traité dans le dernier push

fcayre commented on commown_self_troubleshooting/data/mail_templates.xml:
Je ne comprends pas non plus ce commit : il n'y a pas de mail automatique finalement ?!

loganSIO replied:
Traité dans le dernier push

fcayre commented on product_rental/data/project.xml:
```diff
@@ -6,5 +6,18 @@
 
+    <record id="contract_theft_and_loss_project" model="project.project">
+      <field name="name">Theft and Loss</field>
+      <field name="privacy_visibility">portal</field>
```
privacy_visibility à supprimer ici, pas dans le commit suivant.

loganSIO replied:
Fait!

fcayre reviewed (changes requested):
nickel, il manque juste l'histoire du n° de série, je pense qu'il faut vraiment prévoir un truc si on ne l'a pas.

loganSIO commented:
> nickel, il manque juste l'histoire du n° de série, je pense qu'il faut vraiment prévoir un truc si on ne l'a pas.

@fcayre est-ce que l'ajout d'un texte comme j'ai fait suffit ?

fcayre commented:
> > nickel, il manque juste l'histoire du n° de série, je pense qu'il faut vraiment prévoir un truc si on ne l'a pas.
> 
> @fcayre est-ce que l'ajout d'un texte comme j'ai fait suffit ?

Je veux bien un lien vers le source pour me faciliter un peu la vie...

loganSIO commented:
> > > nickel, il manque juste l'histoire du n° de série, je pense qu'il faut vraiment prévoir un truc si on ne l'a pas.
> > 
> > 
> > @fcayre est-ce que l'ajout d'un texte comme j'ai fait suffit ?
> 
> Je veux bien un lien vers le source pour me faciliter un peu la vie...

https://github.com/commown/commown-odoo-addons/pull/201/commits/c25f8cef9527a16f8c274684c5d6bcc8eaa1f245#diff-e2c5c6f9c901f820a6a75aca51e43b937b61f964dc9d9d626b1169e362af464d

Lignes 131 et 170 @fcayre

fcayre commented:
/ocabot rebase

fcayre commented:
/ocabot rebase

fcayre commented:
/ocabot merge patch
